src/disco/shred/fd_stake_ci.h

```diff
@@ -0,0 +1,80 @@
+#ifndef HEADER_fd_src_app_fdctl_run_tiles_fd_stake_ci_h
+#define HEADER_fd_src_app_fdctl_run_tiles_fd_stake_ci_h
+
+/* fd_stake_ci handles the thorny problem of keeping track of leader
+   schedules and shred destinations, which are epoch specific.  Around
+   epoch boundaries, we may need to query information from the epoch on
+   either side of the boundary. */
+
+#include "fd_shred_dest.h"
+#include "../../flamenco/leaders/fd_leaders.h"
+
+#define MAX_SHRED_DESTS 50000UL
+#define MAX_SLOTS_PER_EPOCH 432000UL
+#define MAX_SHRED_DEST_FOOTPRINT 11244032UL /* == fd_shred_dest_footprint( MAX_SHRED_DESTS ), runtime asserted in new */
+
+struct fd_per_epoch_info_private {
+  /* Epoch, and [start_slot, start_slot+slot_cnt) refer to the time
```

Comment:
This comment is pretty explicit but just because of the confusion this caused us I would add more detail.

I think specifically call out that you do not need to adjust epoch by +1 or +2, even though the leader schedule is based on stakes at a prior epoch.